In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.preprocessing import StandardScaler
from sklearn.neural_network import MLPClassifier
from sklearn.metrics import accuracy_score,confusion_matrix
import warnings
warnings.filterwarnings("ignore")

In [2]:
url = 'https://archive.ics.uci.edu/ml/machine-learning-databases/00236/seeds_dataset.txt'

# Load the dataset
features = ['Area',
            'Perimeter',
            'Compactness',
            'Length of kernel',
            'Width of kernel',
            'Asymmetry coefficient',
            'Length of kernel groove.']

df = pd.read_csv(url, delimiter=r'[\t]+',
                 names=features + ['target'])
X = df.iloc[:, :-1].values
y = df.iloc[:, -1].values

# Split the data into training and testing sets
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# Apply feature scaling to the training and testing data
scaler = StandardScaler()
X_train_scaled = scaler.fit_transform(X_train)
X_test_scaled = scaler.transform(X_test)

# Create an MLP classifier with default hyperparameters
clf = MLPClassifier(random_state=42)

# Fit the classifier to the training data
clf.fit(X_train_scaled, y_train)

# Make predictions on the test data
y_pred = clf.predict(X_test_scaled)

# Evaluate the performance of the model
accuracy = accuracy_score(y_test, y_pred)
print(f'Accuracy: {accuracy:.2f}')
print('\n Confusion matrix: \n', confusion_matrix(y_test, y_pred))

Accuracy: 0.88

 Confusion matrix: 
 [[ 9  0  2]
 [ 0 14  0]
 [ 3  0 14]]


The MLP model was trained on a wheat seeds dataset and achieved an accuracy of 0.88. The confusion matrix shows the breakdown of predicted labels compared to the actual labels.

Looking at the matrix, we can see that the model predicted 9 samples as belonging to class 1 and the actual class was also 1. The model predicted 14 samples as belonging to class 2, and the actual class was also 2. The model predicted 14 samples as belonging to class 3, and the actual class was also 3.

However, there are also some misclassifications. The model predicted 3 samples as belonging to class 1, but the actual class was 3. The model predicted 2 samples as belonging to class 3, but the actual class was 1.

In general, an accuracy of 0.88 indicates that the model is performing well, but it's worth investigating further to see if there are any imbalances in the dataset or if there are any specific classes that the model is having trouble with. Also, when we compare with previous logistic regression the performace gotslightly improved